converters/fromZigbee.js

```diff
@@ -375,6 +403,35 @@ const converters = {
             }
         },
     },
+    lock_pin_code_rep: {
+        cluster: 'closuresDoorLock',
+        type: ['commandGetPinCodeRsp'],
+        convert: (model, msg, publish, options, meta) => {
+            const {data} = msg;
+            let status = '';
+            let pinCodeValue = null;
+            switch (data.userstatus) {
+            case 0:
+                status = 'available';
+                break;
+            case 1:
+                status = 'enabled';
+                if (options && options.read_pin_value && data.pincodevalue) {
```

Comment:
Why do we need to check this option here, if you don't want to read the pin value you never issue the command so it will also not end up here?

Reply:
This still allows someone to see which slots are active, which is needed before setting a code. Cannot set an active code (at least for this device), it must be cleared first.

Reply:
I don't understand how that is related to not putting the `pin_code` in the resulting MQTT message of this. Could you explain this further?

Reply:
Ah - it is indeed putting `pin_code` in the message, but the value is set to null. Perhaps a cleaner result would be to exclude it entirely.

Reply:
So from what I understand, the use case of this option is to prevent the pin code from getting send over MQTT when the user requested the pin code? (as in this case only the `status` will be exposed)

In that case `expose_pin` would be a more logical name in my opinion. (which defaults to `false`).